### PR TITLE
docs(ErrorSummary): write documentation and guidelines

### LINKED ISSUE #1319: ErrorSummary: Write guidelines for usage

### PR DESCRIPTION
**Pull request by Barsnes**
resolves #1319

### CODE REVIEW

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 1]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+`ErrorSummary` skal inneholde alle feilmeldingene som er på siden, så brukere navigere direkte 
+til feilene og manglene.
+
+Du kan lese mer om [oppsummering av flere feilmeldinger på våre nettsider](https://designsystemet.no/monstre/feilmeldinger#oppsummering-av-flere-feilmeldinger).
+
+Det skal være lenker fra `ErrorSummary` til der feilen er på siden. Når brukeren trykker på en av lenkene, 
+skal fokus flyttes til der feilen er.
+
+Lenketeksten i `ErrorSummary` må være identisk med feilen den lenker til.
+
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:80]:
```suggestion

<br/>

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 2]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+### Synlighet
+
+Oppsummeringen skal ikke være synlig uten at brukeren foretar seg noe. Den skal bli synlig først når brukeren har utført en handling om utløser feil.
+
+- Hvis vi validerer fortløpende (feks. ved onBlur på hvert felt), venter vi med å vise ErrorSummary til brukeren prøver å sende inn skjemaet.
+- Det er viktig at komponenten får fokus når den visest, slik at brukeren får med seg at det er feil eller mangler i skjemaet.
+- Feilmeldingene i oppsummeringen skal være formulert lik feilmeldingen ved feltet.
+- Feilmeldingene skal lenke direkte til feltet det gjelder.
+- Hvis feilen gjelder flere felt, lenker vi til det første tilfellet av feilen.
+
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:90]:
```suggestion

<br/>

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 3]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+skal fokus flyttes til der feilen er.
+
+Lenketeksten i `ErrorSummary` må være identisk med feilen den lenker til.
+
+### Synlighet
+
+Oppsummeringen skal ikke være synlig uten at brukeren foretar seg noe. Den skal bli synlig først når brukeren har utført en handling om utløser feil.
+
+- Hvis vi validerer fortløpende (feks. ved onBlur på hvert felt), venter vi med å vise ErrorSummary til brukeren prøver å sende inn skjemaet.
+- Det er viktig at komponenten får fokus når den visest, slik at brukeren får med seg at det er feil eller mangler i skjemaet.
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:86]:
```suggestion
- Det er viktig at komponenten får fokus når den blir vist, slik at brukeren får med seg at det er feil eller mangler i skjemaet.
```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 4]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+Det skal være lenker fra `ErrorSummary` til der feilen er på siden. Når brukeren trykker på en av lenkene, 
+skal fokus flyttes til der feilen er.
+
+Lenketeksten i `ErrorSummary` må være identisk med feilen den lenker til.
+
+### Synlighet
+
+Oppsummeringen skal ikke være synlig uten at brukeren foretar seg noe. Den skal bli synlig først når brukeren har utført en handling om utløser feil.
+
+- Hvis vi validerer fortløpende (feks. ved onBlur på hvert felt), venter vi med å vise ErrorSummary til brukeren prøver å sende inn skjemaet.
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:85]:
```suggestion
- Hvis vi validerer fortløpende (feks. ved `onBlur` på hvert felt), venter vi med å vise ErrorSummary til brukeren prøver å sende inn skjemaet.
```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 5]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
 
 <Canvas of={ErrorSummaryStories.ShowHide} />
 
+## Retningslinjer
+
+Brukes til å vise en feiloppsummering når det er feil i noe brukeren har tastet inn, også hvis det er kun én feil på siden.
+
+Slik fungerer det
+- flytt fokuset på tastaturet til `ErrorSummary`
+- bruk overskriften “det er oppstått et problem”
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:52]:
```suggestion
- overskriften “For å gå videre må du rette opp følgende feil” brukes
```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 6]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+- lag lenke til hvert av svarene som har valideringsfeil
+- forsikre deg om at feilmeldingen i `ErrorSummary` er formulert likt som de som vises ved siden av innspillene med feil
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:54]:
```suggestion
- lenkene i oppsummeringen stemmer med feilmeldingene de lenker til
```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 7]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+
+Passer til å
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:56]:
```suggestion
 
```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 8]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+Brukes til å vise en feiloppsummering når det er feil i noe brukeren har tastet inn, også hvis det er kun én feil på siden.
+
+Slik fungerer det
+- flytt fokuset på tastaturet til `ErrorSummary`
+- bruk overskriften “det er oppstått et problem”
+- lag lenke til hvert av svarene som har valideringsfeil
+- forsikre deg om at feilmeldingen i `ErrorSummary` er formulert likt som de som vises ved siden av innspillene med feil
+
+Passer til å 
+- oppsummere feil for å gi brukeren oversikt over alle feilene, før feilmeldingsskjemaet sendes inn
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:57]:
```suggestion

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 9]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+- gi brukerne muligheten til å hoppe direkte til feltene med feil i skjemaet
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:58]:
```suggestion

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 10]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+
+<br />
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:60]:
```suggestion

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 11]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+
+Passer ikke til å
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:62]:
```suggestion

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 12]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+- forsikre deg om at feilmeldingen i `ErrorSummary` er formulert likt som de som vises ved siden av innspillene med feil
+
+Passer til å 
+- oppsummere feil for å gi brukeren oversikt over alle feilene, før feilmeldingsskjemaet sendes inn
+- gi brukerne muligheten til å hoppe direkte til feltene med feil i skjemaet
+
+<br />
+
+Passer ikke til å 
+- vise lange, tekniske feilkoder til brukerne.
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:63]:
```suggestion

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 13]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+Det er fortsatt viktig at `ErrorSummary` får fokus ved submit slik at brukeren får med seg at det er feil eller mangler i skjemaet.
 
 <Canvas of={ErrorSummaryStories.ShowHide} />
 
+## Retningslinjer
+
+Brukes til å vise en feiloppsummering når det er feil i noe brukeren har tastet inn, også hvis det er kun én feil på siden.
+
+Slik fungerer det
+- flytt fokuset på tastaturet til `ErrorSummary`
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:51]:
```suggestion
- brukerne kan gå direkte til feilene de må rette opp, og fokuset blir flyttet dit
```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 14]
```diff
@@ -24,19 +27,75 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+- bruk overskriften “det er oppstått et problem”
+- lag lenke til hvert av svarene som har valideringsfeil
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:53]:
```suggestion

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 15]
```diff
@@ -24,19 +27,77 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+- Hvis feilen gjelder flere felt, lenker vi til det første tilfellet av feilen.
+
+<br/>
+
+## Tilgjengelighet
+
+`ErrorSummary` har `aria-role='alert'` som standard. Dette betyr at skjermlesere vil lese opp innholdet i `ErrorSummary` automatisk når den vises.
+Alle punkt i listen er lenker, og oppfører seg som en lenke.
+
+<br />
```
Comment (mrosvik) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:98]:
```suggestion

** Pass på at**
- Når oppsummeringen blir synlig, flyttes fokus dit
- Når brukeren trykker på en lenke, flyttes fokus til stedet hvor feilen eller mangelen oppstod

```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 16]
```diff
@@ -24,19 +27,81 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+
+<br/>
+
+## Tilgjengelighet
+
+`ErrorSummary` har `aria-role='alert'` som standard. Dette betyr at skjermlesere vil lese opp innholdet i `ErrorSummary` automatisk når den vises.
+Alle punkt i listen er lenker, og oppfører seg som en lenke.
+
+
+** Pass på at**
```
Comment (mimarz) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:99]:
```suggestion
**Pass på at**
```

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx [hunk 17]
```diff
@@ -24,19 +27,81 @@ import { ErrorSummary } from '@digdir/designsystemet-react';
...
+- Når oppsummeringen blir synlig, flyttes fokus dit
+- Når brukeren trykker på en lenke, flyttes fokus til stedet hvor feilen eller mangelen oppstod
+
+<br />
+
 ### `ErrorSummary.Item`
 
 <ArgTypes of={ErrorSummary.Item} />
```
Comment (mimarz) [inline packages/react/src/components/ErrorSummary/ErrorSummary.mdx:107]: Maybe move this up top close to ErrorSummary preview? Seems weird with it just dangling at the very end

**Review by mimarz — APPROVED** [review]
Looks good, found a md syntax error, added suggestion.

File: packages/react/src/components/ErrorSummary/ErrorSummary.mdx (hunk 17 above)
Comment (Barsnes) [reply packages/react/src/components/ErrorSummary/ErrorSummary.mdx:107]: I think we need to look at this across our documentation later, other components also have this at the bottom

**Pull request merged**